functions-and-scope.js

```diff
@@ -1,39 +1,116 @@
 // * Hoe kan ik iedere waarde van de array checken op deze conditie?
 // * Hoe zorg ik ervoor dat dit ook werkt wanneer de array 100 entries bevat?
 // * Hoe zorgt ik ervoor dat wanneer ik een cijfer tegenkom die aan de conditie voldoet, ik dit ergens kan bijhouden?
 // Log het antwoord in de terminal.
 
 // ---- Verwachte uitkomst: 6
-
+//===========================================
+// console.log('Opdr.1a');
+// console.log("// ---- Verwachte uitkomst: 6");
+// {
+//     console.log(cumLaude1());
+//     function cumLaude1(){
+//         let amountCumLaude1 = 0;
+//         for (let i= 0; i < grades.length; i++) {
+//             if (grades[i] >= 8){
+//                 amountCumLaude1++;
+//             }
+//         }
+//         return amountCumLaude1;
+//     }
+// }
+// console.log('Eind Opdr.1a\n');
+//======================================================================================================================
 
 /*  1b: Omschrijven tot een herbruikbare functie   */
-// Schrijf een functie genaamd cumLaude, die een array van cijfers verwacht (zoals grades) en het aantal Cum laude studenten teruggeeft. Gebruik hiervoor jouw antwoord van 1a.
-// Zorg ervoor dat jouw functie ook werkt als we een andere array met eindcijfers willen checken, zoals bijvoorbeeld: [6, 4, 5] of [8, 9, 4, 6, 10].
+// Schrijf een functie genaamd cumLaude, die een array van cijfers verwacht (zoals grades) en het aantal Cum laude
+// studenten teruggeeft. Gebruik hiervoor jouw antwoord van 1a.
+// Zorg ervoor dat jouw functie ook werkt als we een andere array met eindcijfers willen checken, zoals
+// bijvoorbeeld: [6, 4, 5] of [8, 9, 4, 6, 10].
 // Log het antwoord in de terminal.
 
 // ---- Verwachte uitkomsten:
 // cumLaude(grades) geeft 6
 // cumLaude([6, 4, 5]) geeft 0
 // cumLaude([8, 9, 4, 6, 10]) geeft 3
+//===========================================
+{
+    console.log('Opdr.1b');
+
+    console.log("// ---- Verwachte uitkomsten:\n" +
+        "// cumLaude(grades) geeft 6\n" +
+        "// cumLaude([6, 4, 5]) geeft 0\n" +
+        "// cumLaude([8, 9, 4, 6, 10]) geeft 3");
+
+
+    function cumLaude2(arrGrades) {
+        console.log(arrGrades);
+        let amountCumLaude2 = 0;
+        for (let i = 0; i < arrGrades.length; i++) {
+            // console.log(amountCumLaude2)
+            if (grades[i] >= 8) {
+                amountCumLaude2++;
+            }
+        }
+        return amountCumLaude2;
+    }
+
+    const diplomasOne = cumLaude2(grades);
+    const diplomasTwo = cumLaude2([6, 4, 5]);
+    const diplomasThree = cumLaude2([8, 9, 4, 6, 10]);
+
+    console.log(diplomasOne, diplomasTwo, diplomasThree);
+
+    console.log('Eind Opdr.1b\n');
+}
+//======================================================================================================================
+
+function cumLaude(gradesArray) {
+    let count = 0;
+
+    for (let i = 0; i < gradesArray.length; i++) {
+        if (gradesArray[i] >= 8) {
+            count = count + 1;
```

Comment:
Dit kun je nog wat korter schrijven
```suggestion
            count++;
```

Reply:
Ja.. Begrijp ik .. Ga ik voortaan toepassen inderdaad..